api/src/main/java/org/pmiops/workbench/compliancetraining/ComplianceTrainingServiceImpl.java

```diff
@@ -68,6 +76,24 @@ public DbUser syncComplianceTrainingStatus()
     return syncComplianceTrainingStatus(user, Agent.asUser(user));
   }
 
+  @Override
+  public boolean useAbsorb() {
+    var featureFlagEnabled = configProvider.get().absorb.enabledForNewUsers;
+    var userHasUsedMoodle =
+        userAccessModuleDao.getAllByUser(userProvider.get()).stream()
+            .anyMatch(
+                uam ->
+                    complianceTrainingVerificationDao
+                            .getByUserAccessModule(uam)
+                            .map(
+                                DbComplianceTrainingVerification
+                                    ::getComplianceTrainingVerificationSystem)
+                            .orElse(null)
+                        == DbComplianceTrainingVerification.DbComplianceTrainingVerificationSystem
+                            .MOODLE);
+    return featureFlagEnabled && !userHasUsedMoodle;
```

Comment:
What about a potential future situation something like RT training = Moodle and CT training = Absorb.  Is this logic still correct?  I'd like to see a test demonstrating that.

Reply:
See https://github.com/all-of-us/workbench/pull/8006 👍